Text/Megaparsec.hs

```diff
@@ -329,10 +330,12 @@ pAp m k = ParsecT $ \s cok cerr eok eerr ->
   in unParser m s mcok cerr meok eerr
 {-# INLINE pAp #-}
 
+-- | 'A.empty' will return a parser that __fails__ and consumes no input.
 instance (Ord e, Stream s) => A.Alternative (ParsecT e s m) where
   empty  = mzero
   (<|>)  = mplus
 
+-- | See comment for 'pure'.
```

Comment:
I think it's better to be a bit repetitive here (if we start to give this sort of clarification):

> 'return' returns a parser that __succeeds__ without consuming input.